Dockerfile

```diff
@@ -0,0 +1,123 @@
+FROM python:3.11-slim-bookworm AS base
+
+ARG ENV=dev
+
+# Create the user in the base stage, so all later stages inherit the user setup.
+ARG UID=1000
+ARG GID=1000
+
+# Create our users here in the last layer or else it will be lost in the previous discarded layers
+# Create a system group named "app_user" with the -r flag
+RUN groupadd -g ${GID} -o app
+RUN useradd -m -d /app -u ${UID} -g ${GID} -o -s /bin/bash app
+
+ENV POETRY_HOME="/opt/poetry"
+ENV PATH="$POETRY_HOME/bin:$PATH"
+
+RUN apt-get update -y \
+    && apt-get upgrade -y \
+    && apt-get install -y --no-install-recommends \
+    build-essential \
+    curl \
+    pkgconf \
+    default-libmysqlclient-dev \
+    openssh-client \
+    git \
+    gcc
+
+RUN curl -sSL https://install.python-poetry.org | python3 -
+
+WORKDIR /app
+COPY --chown=${UID}:${GID} ./pyproject.toml ./poetry.lock /app/
+# Copy the common libraries
+COPY --chown=${UID}:${GID} ./libs ../libs/
+
+# Installing dependencies
+RUN if [ "${ENV}" = "dev" ]; then \
+    echo "Installing dev dependencies" && \
+    poetry config virtualenvs.create false && \
+    poetry install --no-root --with dev; \
+    else \
+    echo "Skipping dev dependencies" && \
+    poetry config virtualenvs.create false && \
+    poetry install --no-root --without dev && rm -rf ${POETRY_CACHE_DIR}; \
+    fi
+
+# Builder ht_indexer stage
+FROM base AS ht_indexer
+
+# document_generator_services uses /sdr folder to read zip and mets files
+RUN for i in $(seq 1 24); do ln -s /sdr/$i /sdr$i; done
+
+WORKDIR /app
+
+# Clean up
+RUN apt-get update \
+    && apt-get -y autoremove \
+    && apt-get -y clean \
+    && rm -rf /var/lib/apt/lists/*
+
+# Copy the installed packages from the builder image
+COPY --chown=${UID}:${GID} --from=base /usr/local/lib/python3.11/site-packages /usr/local/lib/python3.11/site-packages
```

Comment:
We're already using `FROM base`, so wouldn't these already be in place? Was the intention to use `[FROM python:](python:3.11-slim-bookworm)` -- that is, build/install all the packages in the `base` step, then copy them into the production image without the need for the whole build pipeline and development tools?